Language/Expressions/Constants/Constant_Contexts/constant_context_A03_t01.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let e be an expression; e occurs in a constant context iff one of
+/// the following applies:
+/// ...
+/// • e is an actual argument in an expression derived from
+///   ⟨constObjectExpression⟩.
+///
+/// @description Checks that an expression `e` is a constant expression if `e`
+/// is an actual argument in an expression derived from ⟨constObjectExpression⟩
+/// @author sgrekhov22@gmail.com
+
+import '../../../../Utils/expect.dart';
+
+class C {
+  final val;
+  const C(this.val);
+}
+
+main() {
+  const c = const C(1);
+  Expect.identical(c, const C(1));
```

Comment:
So the question is whether `1` is treated as a constant expression (that is: whether or not it's canonicalized). But values of the type `int` are _always_ canonicalized, there is no way to allocate a new instance of `int` with the value 7, we can just obtain the unique one that everybody else will also obtain if they need 7.

So the argument to the constructor `C` should probably be `D()` where `D` is some other class that allows for constant as well as non-constant creation (or it could be `[]`, just so we don't have to create an extra class).

Reply:
Changed to `[]`